src/models/event-timeline-set.js

```diff
@@ -576,7 +576,7 @@ EventTimelineSet.prototype.addEventToTimeline = function(event, timeline,
 EventTimelineSet.prototype.handleRemoteEcho = function(localEvent, oldEventId,
                                                         newEventId) {
     // XXX: why don't we infer newEventId from localEvent?
-    const existingTimeline = this._eventIdToTimeline[oldEventId];
+    const existingTimeline = this._eventIdToTimeline[oldEventId] || this._eventIdToTimeline[newEventId];
     if (existingTimeline) {
         delete this._eventIdToTimeline[oldEventId];
         this._eventIdToTimeline[newEventId] = existingTimeline;
```

Comment:
hmm, in the light of the new day I just realized that in case `this._eventIdToTimeline[newEventId]` we would just need to do nothing. Like, the delete operation won't do anything & L582 doesn't change anything either. Maybe it would be better to handle this case separately?